x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/analyze/BlobAnalyzeAction.java

```diff
@@ -329,12 +329,22 @@ public StreamInput streamInput() throws IOException {
                     };
                     if (atomic) {
                         try {
-                            blobContainer.writeBlobAtomic(
-                                OperationPurpose.REPOSITORY_ANALYSIS,
-                                request.blobName,
-                                bytesReference,
-                                failIfExists
-                            );
+                            if (random.nextBoolean()) {
+                                blobContainer.writeBlobAtomic(
+                                    OperationPurpose.REPOSITORY_ANALYSIS,
+                                    request.blobName,
+                                    bytesReference,
+                                    failIfExists
+                                );
+                            } else {
+                                blobContainer.writeBlobAtomic(
+                                    OperationPurpose.REPOSITORY_ANALYSIS,
+                                    request.blobName,
+                                    bytesReference.streamInput(),
+                                    bytesReference.length(),
+                                    failIfExists
+                                );
+                            }
```

Comment:
Nit: I am not sure whether this randomization is really necessary. This is not a test and we don't randomize for `writeBlob` a few lines below?

Reply:
Ah yes! Will revert.